### ISSUE TITLE
IOperation - IOperation API for BoundMethodGroup and BoundPropertyGroup

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#19965]
Would be valuable for things like INameOfExpression: https://github.com/dotnet/roslyn/issues/19954

Probably also want this for IBadInvocationExpression where the member group can specify the members that the compiler considered.

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/analyzer-ioperation

**Comment by jinujoseph** [comment]
No.  Wed on't need this for V1.  However, whoever does Nameof needs to make sure it exposes the list of members that were bound.

**Comment by mavasani** [comment]
I think it is weird to expose members within a nameof expression - in fact the BoundNameOfOperator node doesn't even include any BoundMethodGroup or BoundPropertyGroup: http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Generated/BoundNodes.xml.Generated.cs,985506451dde049a.

Instead a better alternative is to just expose the nameof argument as an IOperation, and in future expose these groups. Until then, the recommendation across our entire API surface should be to revert to semantic model to get bound method and property groups.

**Comment by CyrusNajmabadi** [comment]
> I think it is weird to expose members within a nameof expression 

I don't think it's weird.  It's what the entity inside the nameof ends up meaning semantically.

> Until then, the recommendation across our entire API surface should be to revert to semantic model to get bound method and property groups.

This is undesirable because now you have to go back to syntax, and handle the syntax for each language separately just to ask a question that can be answered at our common ioperation layer.

That said, if this is hard to provide (due to how the bound nodes are structured), then i think it's ok if we punt on having the INameOfOperation actually expose any additional information.  We can design and add such a thing later.

**Comment by jinujoseph** [comment]
**Design Team Decision**

Lets leave this as is for V1 release.

**Comment by mavasani** [comment]
@heejaechang Wants to discuss this in the design meeting.

**Comment by mavasani** [comment]
We need to fix this for 15.5 so that children of bound method and property group are not cut - should be a trivial bug fix.

**Comment by tmat** [comment]
@heejaechang What's the status of this issue? Do you have  anything in progress?

**Comment by heejaechang** [comment]
part of this 
https://github.com/dotnet/roslyn/pull/21857

**Comment by 333fred** [comment]
Closing as a duplicate of a community-driven issue, https://github.com/dotnet/roslyn/issues/76620.

**Comment by CyrusNajmabadi** [comment]
WFM.
